I'm getting a bunch of warnings which may be okay, but then an error which looks unpleasant regarding size inconsistencies.

`ariaExtract.py -f "products/*" -l 'all' -w BB_default -d Download` # using test products from v2v3 test dataset

```
/Users/buzzanga/mambaforge/envs/ARIA-tools/bin/ariaExtract.py:4: DeprecationWarning: pkg_resources is deprecated as an API. See https://setuptools.pypa.io/en/latest/pkg_resources.html
  __import__('pkg_resources').require('ARIAtools==1.1.6')
2024-04-11 08:24:52,753 - ARIAtools.util.vrt - WARNING - Extraction of raider-derived troposphere layers is not possible as specified tropo model name(s) "-tm None" is not valid.
2024-04-11 08:24:53,504 - ARIAtools.util.dem - WARNING - Cannot proceed with VRT format, using ENVI format instead
Downloading glo_90 tiles: 100%|████████████████████████████████████████████████████████████████████████| 20/20 [01:47<00:00,  5.38s/it]
2024-04-11 08:26:44,368 - rasterio._env - WARNING - CPLE_NotSupported in driver ENVI does not support creation option BLOCKYSIZE
2024-04-11 08:26:44,441 - rasterio._env - WARNING - CPLE_NotSupported in driver ENVI does not support creation option BLOCKYSIZE
Generating: solidEarthTide - [===============>        33%                       ] 20220331_20220319    0s /     1s2024-04-11 08:26:44,889 - rasterio._env - WARNING - CPLE_NotSupported in driver ENVI does not support creation option BLOCKYSIZE
2024-04-11 08:26:44,962 - rasterio._env - WARNING - CPLE_NotSupported in driver ENVI does not support creation option BLOCKYSIZE
Generating: solidEarthTide - [======================= 67% =====>                ] 20230829_20230724    1s /     0s2024-04-11 08:26:45,423 - rasterio._env - WARNING - CPLE_NotSupported in driver ENVI does not support creation option BLOCKYSIZE
2024-04-11 08:26:45,491 - rasterio._env - WARNING - CPLE_NotSupported in driver ENVI does not support creation option BLOCKYSIZE
Generating: solidEarthTide - [===============>        33%                       ] 20230829_20230817    1s /     3s
/Users/buzzanga/Software_InSAR/ARIA-tools_git/tools/ARIAtools/util/stitch.py:369: RuntimeWarning: invalid value encountered in multiply
  comb_data = np.empty((n, length, width), dtype=np.float64) * np.nan
Traceback (most recent call last):
  File "/Users/buzzanga/mambaforge/envs/ARIA-tools/bin/ariaExtract.py", line 7, in <module>
    exec(compile(f.read(), __file__, 'exec'))
  File "/Users/buzzanga/Software_InSAR/ARIA-tools_git/tools/bin/ariaExtract.py", line 283, in <module>
    main()
  File "/Users/buzzanga/Software_InSAR/ARIA-tools_git/tools/bin/ariaExtract.py", line 270, in main
    arrshape = ARIAtools.extractProduct.export_products(**export_dict)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/buzzanga/Software_InSAR/ARIA-tools_git/tools/ARIAtools/extractProduct.py", line 1221, in export_products
    ARIAtools.util.vrt.dim_check(ref_arr, prod_arr)
  File "/Users/buzzanga/Software_InSAR/ARIA-tools_git/tools/ARIAtools/util/vrt.py", line 575, in dim_check
    raise Exception(
Exception: Inconsistent product dims between products BB_default/lookAngle/20210330_20200329 and BB_default/unwrappedPhase/20210330_20200329: check respective width (4545, 4379) and height (3821, 3382) and geotrans ((-123.066666666667, 0.000802896956362301, -0.0, 43.0408333333333, -0.0, -0.000737590508592863), (-123.066666666667, 0.000833333333333333, -0.0, 43.0408333333333, -0.0, -0.000833333333333333))
```

--- a/tools/ARIAtools/extractProduct.py
+++ b/tools/ARIAtools/extractProduct.py
@@ -645,7 +645,7 @@ def generate_diff(ref_outname, sec_outname, outname, key, OG_key, tropo_total,
 
 def handle_epoch_layers(
         layers, product_dict, lyr_path, key, sec_key, ref_key, tropo_total,
-        workdir, prog_bar, bounds, dem_bounds, prods_TOTbbox, dem, lat, lon,
+        workdir, prog_bar, bounds, arrres, dem_bounds, prods_TOTbbox, dem, lat, lon,
         mask, outputFormat, verbose, multilooking, rankedResampling,
         num_threads):
     """
@@ -790,7 +790,7 @@ def handle_epoch_layers(
             for j in enumerate(record_epochs):
                 # Interpolate/intersect with DEM before cropping
                 finalize_metadata(
-                    j[1][:-4], bounds, dem_bounds, prods_TOTbbox, dem, lat,
+                    j[1][:-4], bounds, arrres, dem_bounds, prods_TOTbbox, dem, lat,
                     lon, hgt_field, prod_ver_list, mask, outputFormat,
                     verbose=verbose)
 
@@ -867,7 +867,7 @@ def export_product_worker(
 
         # Interpolate/intersect with DEM before cropping
         finalize_metadata(
-            outname, bounds, dem_bounds, prods_TOTbbox, dem_expanded,
+            outname, bounds, arrres, dem_bounds, prods_TOTbbox, dem_expanded,
             lat, lon, hgt_field, product, mask, outputFormatPhys,
             verbose=verbose)
 
@@ -1031,6 +1031,7 @@ def export_products(
     # get bounds
     bounds = ARIAtools.util.shp.open_shp(bbox_file).bounds
     lyr_input_dict['bounds'] = bounds
+    lyr_input_dict['arrres'] = arrres
     if dem_expanded is not None:
         dem_gt = dem_expanded.GetGeoTransform()
         dem_bounds = [
@@ -1275,7 +1276,7 @@ def export_products(
     return retval
 
 
-def finalize_metadata(outname, bbox_bounds, dem_bounds, prods_TOTbbox, dem,
+def finalize_metadata(outname, bbox_bounds, arrres, dem_bounds, prods_TOTbbox, dem,
                       lat, lon, hgt_field, prod_list, mask=None,
                       outputFormat='ENVI', verbose=None, num_threads='2'):
     """Interpolate and extract 2D metadata layer.
@@ -1283,9 +1284,9 @@ def finalize_metadata(outname, bbox_bounds, dem_bounds, prods_TOTbbox, dem,
     3D layers with a DEM.
     Lat/lon arrays must also be passed for this process.
     """
-    arrshape = [dem.RasterYSize, dem.RasterXSize]
+    #arrshape = [dem.RasterYSize, dem.RasterXSize]
     ref_geotrans = dem.GetGeoTransform()
-    arrres = [abs(ref_geotrans[1]), abs(ref_geotrans[-1])]
+    dem_arrres = [abs(ref_geotrans[1]), abs(ref_geotrans[-1])]
 
     # load layered metadata array
     tmp_name = outname + '.vrt'
@@ -1375,16 +1376,18 @@ def finalize_metadata(outname, bbox_bounds, dem_bounds, prods_TOTbbox, dem,
     data_array_nodata = data_array.GetRasterBand(1).GetNoDataValue()
     gdal_warp_kwargs = {
         'format': outputFormat, 'cutlineDSName': prods_TOTbbox,
-        'outputBounds': bbox_bounds, 'dstNodata': data_array_nodata,
-        'xRes': arrres[0], 'yRes': arrres[1], 'targetAlignedPixels': True,
+        'outputBounds': dem_bounds, 'dstNodata': data_array_nodata,
+        'xRes': dem_arrres[0], 'yRes': dem_arrres[1], 'targetAlignedPixels': True,
         'multithread': True, 'options': [f'NUM_THREADS={num_threads}']}
     warp_options = osgeo.gdal.WarpOptions(**gdal_warp_kwargs)
     osgeo.gdal.Warp(tmp_name + '_temp', tmp_name, options=warp_options)
 
     # Adjust shape
     gdal_warp_kwargs = {
-        'format': outputFormat, 'height': arrshape[0], 'width': arrshape[1],
-        'options': [f'NUM_THREADS={num_threads}']}
+        'format': outputFormat, 'cutlineDSName': prods_TOTbbox,
+        'outputBounds': bbox_bounds, 'dstNodata': data_array_nodata,
+        'xRes': arrres[0], 'yRes': arrres[1], 'targetAlignedPixels': True,
+        'multithread': True, 'options': [f'NUM_THREADS={num_threads}']}
     warp_options = osgeo.gdal.WarpOptions(**gdal_warp_kwargs)
     osgeo.gdal.Warp(outname, tmp_name + '_temp', options=warp_options)
 

--- a/tools/ARIAtools/util/url.py
+++ b/tools/ARIAtools/util/url.py
@@ -16,12 +16,12 @@ def url_versions(urls, user_version, wd):
     Uses the the latest if user_version is None else use specified ver.
     """
     if user_version is not None and str(user_version).lower() != 'all':
-        # dummy-proof if version entered as digit, as opposed to 2_X_X
-        if len(user_version) != 5:
-            raise Exception(f'Input version {user_version} not in format 2_X_X'
-                            'as expected')
+        user_version = user_version.lstrip('v')
+        if not user_version[0].isdigit():
+            raise Exception(f'Input version {user_version} not in format X* '
+                            'as expected (e.g., 3, 3_0_0)')
         LOGGER.debug(f'Only using products version: {user_version}')
-        urls_final = [url for url in urls if user_version in url]
+        urls_final = [url for url in urls if f'-v{user_version}' in url]
         if not urls_final:
             raise Exception(
                 f'No products with user specified version: {urls_final}')

--- a/tools/ARIAtools/util/url.py
+++ b/tools/ARIAtools/util/url.py
@@ -6,6 +6,8 @@
 #
 # ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 import os
+import logging
+LOGGER = logging.getLogger(__name__)
 
 # Grab older version products if specified.
 def url_versions(urls, user_version, wd):
@@ -15,10 +17,10 @@ def url_versions(urls, user_version, wd):
     """
     if user_version is not None and str(user_version).lower() != 'all':
         # dummy-proof if version entered as digit, as opposed to 2_X_X
-        if len(user_version) == 1:
-            raise Exception('Input version {} not in format 2_X_X'
-                            'as expected'.format(user_version))
-        print(f'Only using products version: {user_version}')
+        if len(user_version) != 5:
+            raise Exception(f'Input version {user_version} not in format 2_X_X'
+                            'as expected')
+        LOGGER.debug(f'Only using products version: {user_version}')
         urls_final = [url for url in urls if user_version in url]
         if not urls_final:
             raise Exception(

